### PR TITLE
Add project modules-test-helper

### PR DESCRIPTION
**Pull request by bertfrees**
with JUnit base test classes to reduce boilerplate in tests.

This PR is part of  https://github.com/daisy/pipeline/pull/504

Depends on:
- [ ] https://github.com/daisy/pipeline-build-utils/pull/32

### CODE REVIEW

File: modules-test-helper/src/main/java/org/daisy/pipeline/junit/AbstractTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+				felixDeclarativeServices(),
+				thisBundle(),
+				junitBundles(),
+				systemPackage("javax.xml.stream;version=\"1.0.1\""),
+				mavenBundle("org.daisy.pipeline.build:modules-test-helper:?"),
+				mavenBundlesWithDependencies(
+					testDependencies,
+					// logging
+					logbackClassic(),
+					mavenBundle("org.slf4j:jcl-over-slf4j:1.7.2"))
```
Comment (bertfrees) [inline modules-test-helper/src/main/java/org/daisy/pipeline/junit/AbstractTest.java]: Required by httpclient (Calabash dependency). The real fix is to add this bundle to the runtime dependencies of Calabash.

**Comment by bertfrees** [comment]
Good point. Yes I guess it would be possible. We can simply override or exclude the xproc-engine-daisy-pipeline dependency in the calling pom, or simply remove the dependency from modules-test-helper altogether.

Another problem is that we need to release a new version of xproc-engine-daisy-pipeline for each version of the framework. This is a bit harder to solve because here the framework dependency is a bom import: https://github.com/daisy/xproc-maven-plugin/blob/75f8b08/xproc-engine-daisy-pipeline/pom.xml#L20.

**Comment by rdeltour** [comment]
> We can simply override or exclude the xproc-engine-daisy-pipeline dependency in the calling pom, or simply remove the dependency from modules-test-helper altogether.

👍 for whatever you think is the easiest (maybe override+exclude, to make it work out of the box by default?)

> Another problem is that we need to release a new version of xproc-engine-daisy-pipeline for each version of the framework. This is a bit harder to solve because here the framework dependency is a bom import

Yeah. I've always be niggled by that, but really can't figure what we could improve there 😕

**Comment by bertfrees** [comment]
> This is a bit harder to solve because here the framework dependency is a bom import

Here we could of course also just override the framework dependencies.

The problem with this is that you make the amount of Maven boilerplate bigger again. If you look at [the POM of xproc-engine-daisy-pipeline](https://github.com/daisy/xproc-maven-plugin/blob/75f8b08/xproc-engine-daisy-pipeline/pom.xml#L40) you can see that there are 4 (direct) dependencies: xproc-api, framework-core, common-utils and calabash-adapter. The nice thing about xproc-engine-daisy-pipeline was that you only had to specify one dependency. I guess common-utils and xproc-api could be removed because they are both dependencies of framework-core. calabash-adapter could become a runtime dependency of framework-core since it's our only XProc engine anyway.

**Comment by bertfrees** [comment]
Fixed in 7718890

**Pull request closed without merging**